srtcore/core.cpp

```diff
@@ -5746,9 +5745,10 @@ int CUDT::sendmsg2(const char *data, int len, SRT_MSGCTRL& w_mctrl)
             {
                 const steady_clock::time_point exptime = steady_clock::now() + milliseconds_from(m_iSndTimeOut);
 
-                while (stillConnected() && sndBuffersLeft() < minlen && m_bPeerHealth && exptime > steady_clock::now())
+                while (stillConnected() && sndBuffersLeft() < minlen && m_bPeerHealth)
                 {
-                    sendcond.wait_for(milliseconds_from(m_iSndTimeOut));
+                    if (!sendcond.wait_until(exptime))
+                        break;
```

Comment:
The same happens here. Spurious wake up and condition is net rechecked.
I **don't think** `break` should be added here.

Reply:
This `break` is intended to be executed unless the exit from `wait_until` was due to getting a signal (or if it was spurious) - in which case the loop and its condition is repeated.